.github/workflows/build_base.yaml

```diff
@@ -0,0 +1,148 @@
+# manually input version no.s here
+# will check in workflows to obtain 
+# this info from other files
+#
+
+name: Build base image
+
+on:
+  workflow_call:
+    inputs:
+      cmake_version:
+        type: string
+        required: false
+        default: 3.17.0
+      gcc_version:
+        type: string
+        required: false
+        default: 9
+      clang_version:
+        type: string
+        required: false
+        default: 11
+      ubuntu_version:
+        type: string
+        required: false
+        default: 20.04
+      madness_version:
+        type: string
+        required: false
+        default: 997e8b458c4234fb6c8c2781a5df59cb14b7e700
+      catch2_version:
+        type: string
+        required: false
+        default: 2.13.8
+      spdlog_version:
+        type: string
+        required: false
+        default: ad0e89cbfb4d0c1ce4d097e134eb7be67baebb36
+      cereal_version:
+        type: string
+        required: false
+        default: 1.3.0
+  workflow_dispatch:
+    inputs:
+      cmake_version:
```

Comment:
Fine for now, but repeating the inputs like this seems error prone.

Reply:
Currently I set up a workflow_dispatch event simply for testing purpose (I have tested the workflow in my own organization, but want to make sure everything is alright before I replace the current CI with this one). I will remove the workflow_dispatch event as well as its inputs in the next PR.

Reply:
The condition for merging this PR should be that it works. I also don't understand why the traditional CI didn't run since I don't see any deleted files.

Reply:
We don't need to release ParallelZone images with this PR, but we do need to verify that the workflow can build ParallelZone in a container and can then run the tests in either that container or a container based on it. I don't understand the concerns with the multiple steps/tokens; whatever steps are needed for the CI to work need to be in place.

Reply:
> We don't need to release ParallelZone images with this PR, but we do need to verify that the workflow can build ParallelZone in a container and can then run the tests in either that container or a container based on it. I don't understand the concerns with the multiple steps/tokens; whatever steps are needed for the CI to work need to be in place.

Ok, I can add more workflows into this PR. One thing to confirm: at this time we delay the implementation of the master/dev model, so I can remove the integration test step (currently we don't have realistic integration tests), right?

Reply:
I guess you can remove the integration step, I'm not sure what that step entails at this point.

Reply:
> I guess you can remove the integration step, I'm not sure what that step entails at this point.

I will reserve the place for integration test in the template reusable workflow in .github, but do not call it from individual repos at this time. I will check in the template workflow to .github first, then add the calling workflow to this PR.